src/os/bluestore/bluestore_tool.cc

```diff
@@ -219,6 +227,62 @@ void inferring_bluefs_devices(vector<string>& devs, std::string& path)
   }
 }
 
+void bluefs_import(string input_file,
+                   CephContext *cct,
+                   const string& path,
+                   const vector<string>& devs)
+{
+    int r;
+    std::ifstream f;
+    f.open(input_file.c_str(), std::ifstream::binary);
+    if (!f.is_open()) {
+      r = -errno;
+      cerr << "open " << input_file.c_str() << " failed: " << cpp_strerror(r) << std::endl;
+      exit(EXIT_FAILURE);
+    }
+    struct stat file_stat;
+    r = ::stat(input_file.c_str(), &file_stat);
+    if (r < 0) {
+      cerr << "stat " << input_file.c_str() << " failed: " << cpp_strerror(r) << std::endl;
+      exit(EXIT_FAILURE);
+    }
+
+    fs::path file_path(input_file);
+    const string file_name = file_path.filename();
+
+    std::unique_ptr<BlueFS> bs(new BlueFS(cct));
+    bs = std::move(std::unique_ptr<BlueFS>(open_bluefs(cct, path, devs)));
+
+    const string dir = "db";
+    BlueFS::FileWriter *h;
+    uint64_t dsize = 0;
+    r = bs->stat(dir, file_name, &dsize, NULL);
+    if (r != 0){
+      cerr << "Get " << dir << "/" << file_name << " stat failed: " << cpp_strerror(r) << std::endl;
+      exit(EXIT_FAILURE);
+    }
+
+    r = bs->open_for_write(dir, file_name, &h, true);
+    if (r < 0) {
+      cerr << "Open " << dir << "/" << file_name << " failed: " << cpp_strerror(r) << std::endl;
+      exit(EXIT_FAILURE);
+    }
+    if (uint64_t(file_stat.st_size) < dsize){
```

Comment:
might want to use [file_size](https://en.cppreference.com/w/cpp/filesystem/file_size) ?